registry/repository.go

```diff
@@ -24,6 +24,11 @@ type RepositoryClient struct {
 	remote.Repository
 }
 
+type SignatureManifest struct {
+	Blob        artifactspec.Descriptor
```

Comment:
Can we keep the blob be `notation.Descriptor` rather than `artifactSpec.Descriptor`?